numpy/core/getlimits.py

```diff
@@ -9,19 +9,198 @@
+
+
+# Known parameters for float32.
+# See docstring of MachAr class for description of parameters.
+_f32 = ntypes.float32
+_float32_ma = MachArLike(_f32,
+                         machep=-23,
+                         negep=-24,
+                         minexp=-126,
+                         maxexp=128,
+                         it=23,
+                         iexp=8,
+                         ibeta=2,
+                         irnd=5,
+                         ngrd=0,
+                         eps=exp2(_f32(-23)),
+                         epsneg=exp2(_f32(-24)),
+                         huge=_get_next(ntypes.float32, numeric.inf, 0),
+                         tiny=exp2(_f32(-126)))
+
+# Known parameters for float64
+_f64 = ntypes.float64
+_float64_ma = MachArLike(_f64,
+                         machep=-52,
+                         negep=-53,
+                         minexp=-1022,
+                         maxexp=1024,
+                         it=52,
+                         iexp=11,
+                         ibeta=2,
+                         irnd=5,
+                         ngrd=0,
+                         eps=2.0 ** -52.0,
+                         epsneg=2.0 ** -53.0,
+                         huge=_get_next(_f64, numeric.inf, 0),
+                         tiny=2.0 ** -1022.0)
+
+# Known parameters for float80
+_ld = ntypes.longdouble
+_float80_ma = MachArLike(_ld,
+                         machep=-63,
+                         negep=-64,
+                         minexp=-16382,
+                         maxexp=16384,
+                         it=63,
+                         iexp=15,
+                         ibeta=2,
+                         irnd=5,
+                         ngrd=0,
+                         eps=exp2(_ld(-63)),
+                         epsneg=exp2(_ld(-64)),
+                         huge=_get_next(_ld, numeric.inf, 0),
+                         tiny=exp2(_ld(-16382)))
+
+# Guessed / known parameters for double double; see:
+# https://en.wikipedia.org/wiki/Quadruple-precision_floating-point_format#Double-double_arithmetic
+# These numbers have the same exponent range as float64, but extended number of
+# digits in the significand.
+_float_dd_ma = MachArLike(_ld,
+                          machep=-105,
+                          negep=-106,
+                          minexp=-1022,
+                          maxexp=1024,
+                          it=105,
+                          iexp=11,
+                          ibeta=2,
+                          irnd=5,
+                          ngrd=0,
+                          eps=exp2(_ld(-105)),
+                          epsneg=exp2(_ld(-106)),
+                          huge=_get_next(_ld, numeric.inf, 0),
+                          tiny=exp2(_ld(-1022)))
+
+
+# Key to identify the floating point type is tuple (number of bits in
+# significand (ignoring implicit first digit), maximum exponent, number of
+# bytes).
+_KNOWN_TYPES = {(52, 1024, 8): _float64_ma,
```

Comment:
Here I'd like to eventually get rid of nextafter and just have a big dictionary keyed by byte strings, where the string is generated by something like
```
>>> np.array(1234, dtype='g').tostring()
b'\x00\x00\x00\x00\x00\x00@\x9a\t@\xd73-V\x00\x00'
```
for all the types. I don't know how stable/unique the encoding is, however, and we would need both little and big endian versions. Signed/unsigned characters could be a problem perhaps.